script/tool/test/firebase_test_lab_command_test.dart

```diff
@@ -245,10 +245,10 @@ void main() {
       ]);
 
       processRunner.mockProcessesForExecutable['gcloud'] = <Process>[
-        MockProcess.succeeding(), // auth
-        MockProcess.succeeding(), // config
+        MockProcess(), // auth
+        MockProcess(), // config
```

Comment:
In the future maybe we could make the mock process a bit smarter about arguments too so we aren't relying on the order in which things are called.  This still seems a bit fragile.

Reply:
I'm torn on this; on one hand it's somewhat fragile, on the other the `flutter/flutter` tests used to do process simulation based on mock argument matching and it was *really* painful to get right sometimes, because you had to have correct matchers for a bunch of stuff you probably didn't care about (and it's a different kind of fragile because then, say, re-ordering flags that are irrelevant to the test is test-breaking, unless you have a wildcard system, but wildcards in mock tooling are usually their own can of worms as Mockito demonstrated during the NNBD tranistion). I believe `flutter/flutter` switched to something order-based later.

I want to revisit it at some point (having to use comments in cases like this was a bit of a red flag), but I also want to be careful not to make something that actually makes writing tests substantially harder.